llvm/lib/Target/AMDGPU/SIInstrInfo.cpp

```diff
@@ -8966,10 +8966,10 @@ MachineInstrBuilder SIInstrInfo::getAddNoCarry(MachineBasicBlock &MBB,
                              : RS.scavengeRegisterBackwards(
                                    *RI.getBoolRC(), I, /* RestoreAfter */ false,
                                    0, /* AllowSpill */ false);
-
-  // TODO: Users need to deal with this.
   if (!UnusedCarry.isValid())
-    return MachineInstrBuilder();
+    // We are using the mad_u32_u24 primarily as an add with no carry out
```

Comment:
Braces. But also need to rename this, since it's not valid in the general case.

Reply:
should add new helper for this?